### PR TITLE
Display wordlists

### LINKED ISSUE #2: Create Context Packs
Give the user the ability to add word lists to a context pack

### LINKED ISSUE #3: Fix badges and enable PR checks for better code hub and LGTM

### PR DESCRIPTION
**Pull request by schu4276**
This adds the code base for displaying existing wordlists, and a start on the addition of new wordlists.

### CODE REVIEW

**Review by dahlgrenthomas — DISMISSED** [review]
Looks awesome

File: server/src/main/java/umm3601/contextpack/ContextPackController.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    List<Bson> filters = new ArrayList<>();
+
+    ctx.json(wordlistCollection.find(filters.isEmpty()? new Document() : and(filters))
+    .into(new ArrayList<>()));
+  }
+
+
+  public void addNewWordlist(Context ctx){
+    Wordlist newList = ctx.bodyValidator(Wordlist.class)
+      .check(list -> list.topic.matches("^[a-zA-Z]+$") && list.topic != null)
```
Comment (NicMcPhee) [inline server/src/main/java/umm3601/contextpack/ContextPackController.java]: [As LGTM is telling you](https://lgtm.com/projects/g/UMM-CSci-3601-S21/it-1-purple-tigers/snapshot/a6e85d3f22a65fc5191cd64ca504a6a86184f993/files/server/src/main/java/umm3601/contextpack/ContextPackController.java?sort=name&dir=ASC&mode=heatmap), the order of the tests here is wrong in a meaningful way. You only check for `list.top != null` _after_ you've used it in `list.topic.matches`. That means that if it is `null` you'll get a `NullPointerException` when that guard should have ensured that didn't happen.

**Review by NicMcPhee — CHANGES_REQUESTED** [review]
I didn't do a full review (this is a _huge_ PR, which means no one is probably doing a really careful review), but I did copy over the more serious of the three LGTM warnings. You really do need to address that before you merge this in.

**Comment by schu4276** [comment]
Messy pull request, we are going to reorganize and redo

**Review by NicMcPhee — APPROVED** [review]
Thanks for fixing that.

**Pull request merged**